calico_containers/pycalico/datastore_datatypes.py

```diff
@@ -396,6 +396,14 @@ def __setitem__(self, key, value):
             value = IPNetwork(value)
         if key == "action" and value not in ("allow", "deny"):
             raise ValueError("'%s' is not allowed for key 'action'" % value)
+        if key == "protocol" and value not in ("tcp", "udp", "icmp", None):
+            raise ValueError("'%s' is not allowed for key 'protocol'" % value)
+        if key in ("src_tag", "dst_tag") and re.search('[^a-zA-Z0-9\.\_\-]', value):
+            raise ValueError("'%s' is not allowed for key '%s'" % (value, key))
+        if key in ("src_ports", "dst_ports") and not check_ports(value):
+            raise ValueError("'%s' is not allowed for key '%s'" % (value, key))
+        if key == "icmp_type" and value not in range(256):
```

Comment:
In Python2 I think range() creates a list, so this isn't very efficient.  Correct me if I'm wrong tho.